### DISCUSSION TITLE
How to prove that you have the best model if the models being compared are not tuned？

### DISCUSSION THREAD

**Discussion by huangtao36** [pycaret/pycaret#2649] [Q&A]
I'm confused. compare_models compares the desired models, but it is evaluated without tuning. How can I prove that I got the best model? Is it possible to get a new best model after tuning and re-comparison?

**Comment by Insighttful** [comment]
> Is it possible to get a new best model after tuning and re-comparison?

Yes, it is simply a matter of running `compare_models` on the tuned models and comparing the raw and tuned results. The following exhibits a noticeable increase in R2 for the tuned top models, while also saving you the unneeded computational waste tuning all models.

Ultimately, since all models are part of the experiment 'session', executing `automl` on the experiment will choose from all tuned and untuned models in selecting the best model. While not shown in the example, the automl selection also includes any ensembles you might train on the tuned models.

Using the Pycarat 3 OOP interface:
```python3
from typing import Final
from pycaret.datasets import get_data
from pycaret.regression import RegressionExperiment

# Constants
BEST_N_MODELS: Final[int] = 5
SCORE_METRIC: Final[str] = 'R2'

# Load demo dataset
data = get_data("insurance")

# Start experiment session
exp = RegressionExperiment()

# Set up experiment
exp.setup(data)

# Compare models
exp.compare_models(exclude=['catboost'])
compare_raw = exp.pull()

# Select best n models
best_raw_models = exp.compare_models(n_select=BEST_N_MODELS)

# Tune best raw models
tuned_models = [
    exp.tune_model(
        model,
        choose_better=True,
        early_stopping=True,
        tuner_verbose=False,
    )
    for model in best_raw_models
]

# Compare tuned models
exp.compare_models(tuned_models)
compare_tuned = exp.pull()

# Select best model from raw and tuned
best_model = exp.automl(optimize=SCORE_METRIC)
best_model_stats = exp.pull()
```

### compare_raw
![image](https://github.com/pycaret/pycaret/assets/116014557/311d9fd8-9675-4ffb-82f7-3690913f0e4f)

### compare_tuned
![image](https://github.com/pycaret/pycaret/assets/116014557/b5659d6c-a0db-415a-9192-358f7275f17f)
